packages/react-reconciler/src/ReactFiberHooks.js

```diff
@@ -1524,6 +1535,46 @@ function rerenderTransition(
   return [start, isPending];
 }
 
+function mountOpaqueIdentifier(): OpaqueIDType | void {
+  let hasSetId = false;
+
+  const [id, setId] = mountState(() => {
+    if (getIsHydrating()) {
+      return makeOpaqueHydratingObject(() => {
+        if (!hasSetId) {
+          setId(() => makeClientId());
+          hasSetId = true;
+        }
+      }, getComponentName(currentlyRenderingFiber.type));
```

Comment:
Looks like we're *always* stringifying the component name, even though we *only* use the name for DEV warnings. Seems like we should only call `getComponnetName` in DEV mode, and probably only if we actually need to show a warning?

(So maybe we'd just pass the Fiber or the type?)